scripts/sct_deepseg.py

```diff
@@ -0,0 +1,152 @@
+        description="Segmentation using deep learning.",
+        add_help=None,
+        formatter_class=SmartFormatter,
+        prog=os.path.basename(__file__).strip(".py"))
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        "-i",
+        help="Image to segment.",
+        metavar=Metavar.file)
+
+    seg = parser.add_argument_group('\nMODELS')
+    seg.add_argument(
+        "-m",
+        help="Model to use. For a description of each available model, please write: 'sct_deepseg -list-models'.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-list-models",
+        action='store_true',
+        help="Display a list of available models.")
+    seg.add_argument(
+        "-install-model",
+        help="Install specified model.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-install-default-models",
+        action='store_true',
+        help="Install default models. Note: these models are downloaded during normal SCT installation.")
+    seg.add_argument(
+        "-mpath",
+        help="Path to model, in case you would like to use a custom model. The model folder should follow the "
+             "conventions listed in: URL.",
+        metavar=Metavar.folder)
+
+    misc = parser.add_argument_group('\nPARAMETERS')
+    misc.add_argument(
+        "-thr",
+        type=float,
+        help="Binarize segmentation with specified threshold. Set to 0 for no thresholding (i.e., soft segmentation). "
+             "Default value is model-specific and was set during optimization "
+             "(more info at https://github.com/sct-pipeline/deepseg-threshold).",
+        metavar=float,
+        default=param_default.threshold)
+    misc.add_argument(
+        "-keep-largest",
+        type=bool,
+        help="Remove false negative predictions by only keeping the largest blob.",
+        metavar=bool,
+        default=param_default.keep_largest_object)
+
+    misc = parser.add_argument_group('\nMISC')
+    misc.add_argument(
+        "-o",
+        help="Output segmentation suffix. In case of multi-class segmentation, class-specific suffixes will be added.",
+        metavar=str,
+        default=param_default.output_suffix)
+    misc.add_argument(
+        "-v",
+        type=int,
+        help="Verbose: 0 = no verbosity, 1 = verbose.",
+        choices=(0, 1),
+        default=1)
+    misc.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit")
+
+    return parser
+
+
+def main():
+    param = sct.deepseg.core.ParamDeepseg()
+
+    parser = get_parser()
+    args = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
+    # TODO: instead of assigning each args param, we could pass args while instanciating ParamDeepseg(args), and the
+    #  class would deal with assigning arguments to each field.
```

Comment:
Instead of using `ParamDeepseg`, you could do something like:

```suggestion
    args = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
    args ={k : v for k in vars(args).items() if v is not None }
    # separate out the segmentation `param` args from the top level args
    input = args.pop('i')
    list_models = args.pop('list_models')
    install_model = args.pop('install_model')
    install_default_models = args.pop('install_default_models')
    ...
```

You'd have to remove all the default values/make sure they're set to `None`; `default=param_default.keep_largest_object` is an extra layer of defaults which might accidentally override a model's `metadata` even when the the user doesn't ask for it.

Reply:
actually it fails because v is not defined

Reply:
You have to watch out because the argparse names don't match the postprocess.py names, you have `args.thr` vs `param.threshold` and `args.keep_largest` vs `param.keep_largest_object`; you can use [`dest`](https://docs.python.org/3/library/argparse.html#dest) to fixup the naming, or just change the names of the flags.

Reply:
You could be explicit about which subset are destined for algorithm parameters instead of being using `.pop()` after the fact to pull off the other args:

```
params = ['keep_largest_object', 'fill_holes', 'threshold']
params = {k: vars(args)[k] for k in params if vars(args)[k] is not None}
```

Reply:
Oh sorry!! Let me try that again:

```suggestion
    args = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
    args ={k: v for k, v in vars(args).items() if v is not None}
    # separate out the segmentation `param` args from the top level args
    input = args.pop('i')
    list_models = args.pop('list_models')
    install_model = args.pop('install_model')
    install_default_models = args.pop('install_default_models')
    ...
```

Reply:
@kousu how about not "popping" fields from `args`, and just passing `args` inside `deepseg.core(args)`, because the two first arguments (fname_input and path_model) are already included in args.

Are you worried that we would be passing a bunch of irrelevant fields inside `deepseg.core`?